packages/split-button/src/SplitButton/SplitButton.spec.tsx

```diff
@@ -0,0 +1,221 @@
+import React from 'react';
+import { fireEvent, render } from '@testing-library/react';
+import { axe } from 'jest-axe';
+
+import { MenuItem } from '@leafygreen-ui/menu';
+
+import { MenuItemsType } from './SplitButton.types';
+import { SplitButton } from '.';
+
+const menuTestId = 'lg-split-button-menu';
+
+const getMenuItems = (): MenuItemsType => {
+  return (
+    <>
+      <MenuItem disabled>Disabled Menu Item</MenuItem>
+      <MenuItem description="I am also a description">
+        Menu Item With Description
+      </MenuItem>
+    </>
+  );
+};
+
+const defaultProps = {
+  label: 'Button Label',
+  menuItems: getMenuItems(),
+};
+
+function renderSplitButton(props = {}) {
+  const utils = render(
+    <SplitButton data-testid="split-button" {...defaultProps} {...props} />,
+  );
+  const wrapper = utils.container.firstChild as HTMLElement;
+  const primaryButton = utils.getByTestId('split-button');
+  const menuTrigger = primaryButton.nextSibling as HTMLElement;
+  return {
+    ...utils,
+    primaryButton,
+    menuTrigger,
+    wrapper,
+  };
+}
+
+describe('packages/split-button', () => {
+  describe('a11y', () => {
+    test('does not have basic accessibility issues', async () => {
+      const { container } = renderSplitButton({});
+      const results = await axe(container);
+      expect(results).toHaveNoViolations();
+    });
+  });
+
+  describe('wrapper container', () => {
+    test('renders correct className', () => {
+      const { wrapper } = renderSplitButton({
+        className: 'split-button-class',
+      });
+
+      expect(wrapper.classList.contains('split-button-class')).toBe(true);
+    });
+  });
+
+  describe('primary button', () => {
+    test('renders the correct label', () => {
+      const { primaryButton } = renderSplitButton({});
+
+      expect(primaryButton.textContent).toBe(defaultProps.label);
+    });
+
+    test('is disabled when disabled is true', () => {
+      const { primaryButton } = renderSplitButton({
+        disabled: true,
+      });
+
+      expect(primaryButton.getAttribute('aria-disabled')).toBe('true');
+    });
+
+    test('fires onClick handler once when clicked', () => {
+      const onClick = jest.fn();
+      const { primaryButton } = renderSplitButton({
+        onClick,
+      });
+
+      fireEvent.click(primaryButton);
+      expect(onClick).toHaveBeenCalledTimes(1);
+    });
+
+    test('does not fire onClick handler when disabled', () => {
+      const onClick = jest.fn();
+      const { primaryButton } = renderSplitButton({
+        disabled: true,
+        onClick,
+      });
+
+      fireEvent.click(primaryButton);
+      expect(onClick).toHaveBeenCalledTimes(0);
+    });
+  });
+
+  describe('menu', () => {
```

Comment:
Should test Escape, Enter & Arrow Keys as well

Reply:
Likely just a copy/paste from `Menu`, but we should test that we're not breaking something that's working in Menu